src/ring_theory/graded_algebra/radical.lean

```diff
@@ -0,0 +1,211 @@
+        : begin
+          rw [graded_algebra.proj_apply, alg_equiv.map_mul, graded_algebra.support,
+            graded_algebra.support, direct_sum.coe_mul_apply_submodule],
+          refl,
+        end
+    ... = ∑ ij in (((graded_algebra.support 𝒜 x).product (graded_algebra.support 𝒜 y)).filter
+            (λ (z : ι × ι), z.1 + z.2 = max₁ + max₂)).erase (max₁, max₂),
+            (graded_algebra.proj 𝒜 ij.1 x) * (graded_algebra.proj 𝒜 ij.2 y) +
+          (graded_algebra.proj 𝒜 max₁ x) * (graded_algebra.proj 𝒜 max₂ y)
+        : begin
+          rw finset.sum_erase_add,
+          simp only [finset.mem_filter, finset.mem_product, eq_self_iff_true, and_true],
+          exact ⟨(finset.filter_subset _ _) mem_max₁, (finset.filter_subset _ _) mem_max₂⟩,
+        end,
+
+  have eq₂ : (graded_algebra.proj 𝒜 max₁) x * (graded_algebra.proj 𝒜 max₂) y
+          = graded_algebra.proj 𝒜 (max₁ + max₂) (x * y)
+          - ∑ (ij : ι × ι) in (((graded_algebra.support 𝒜 x).product
+              (graded_algebra.support 𝒜 y)).filter
+              (λ (z : ι × ι), z.fst + z.snd = max₁ + max₂)).erase (max₁, max₂),
+              (graded_algebra.proj 𝒜 ij.fst) x * (graded_algebra.proj 𝒜 ij.snd) y,
+  { rw [eq, eq_sub_iff_add_eq, add_comm], },
+
+  have mem_I :
+    (graded_algebra.proj 𝒜 (max₁, max₂).fst) x * (graded_algebra.proj 𝒜 (max₁, max₂).snd) y ∈ I,
+  { rw eq₂,
+    refine ideal.sub_mem _ hxy (ideal.sum_mem _ (λ z H, _)),
+    rcases z with ⟨i, j⟩,
+    rw finset.mem_erase at H,
+    simp only [not_and, prod.mk.inj_iff, ne.def, finset.mem_filter, finset.mem_product] at H,
+    rcases H with ⟨H₁, ⟨H₂, H₃⟩, H₄⟩,
+    rcases lt_trichotomy i max₁ with h | h | h,
+    { -- in this case `i < max₁`, so `max₂ < j`, so `of A j (y j) ∈ I`
+      have ineq : max₂ < j,
+      { by_contra rid₂, rw not_lt at rid₂,
+        have rid₃ := add_lt_add_of_le_of_lt rid₂ h,
+        conv_lhs at rid₃ { rw add_comm },
+        conv_rhs at rid₃ { rw add_comm },
+        rw H₄ at rid₃,
+        exact lt_irrefl _ rid₃, },
+      have not_mem_j : j ∉ set₂,
+      { intro rid₂,
+        have rid₃ := (finset.max'_lt_iff set₂ set₂_nonempty).mp ineq j rid₂,
+        exact lt_irrefl _ rid₃, },
+      rw set₂_eq at not_mem_j,
+      simp only [not_and, not_not, ne.def, dfinsupp.mem_support_to_fun,
+        finset.mem_filter] at not_mem_j,
+      specialize not_mem_j H₃,
+      apply ideal.mul_mem_left,
+      convert not_mem_j, },
+    { -- in this case `i = max₁`, so `max₂ = j`, contradictory
+      have : j = max₂,
+      { rw h at H₄,
+        exact linear_ordered_cancel_add_comm_monoid.add_left_cancel _ _ _ H₄, },
+      exact false.elim (H₁ h this), },
+    { -- in this case `i > max₁`, so `i < max₁`, so `of A i (x i) ∈ I`
+      have ineq : max₁ < i,
+      { by_contra rid₂, rw not_lt at rid₂,
+        have rid₃ := add_lt_add_of_le_of_lt rid₂ h,
+        conv_lhs at rid₃ { rw linear_ordered_cancel_add_comm_monoid.add_comm },
+        exact lt_irrefl _ rid₃, },
```

Comment:
What time of night did you write this? `ineq` is just the `h` you already have in your goal state 😆